homeassistant/components/screenlogic/climate.py

```diff
@@ -150,14 +151,19 @@ async def async_set_temperature(self, **kwargs: Any) -> None:
         if (temperature := kwargs.get(ATTR_TEMPERATURE)) is None:
             raise ValueError(f"Expected attribute {ATTR_TEMPERATURE}")
 
-        if not await self.gateway.async_set_heat_temp(
-            int(self._data_key), int(temperature)
-        ):
+        try:
+            if not await self.gateway.async_set_heat_temp(
+                int(self._data_key), int(temperature)
+            ):
+                raise ScreenLogicError("Unexpected response")
+            _LOGGER.debug(
+                "Set temperature for body %s to %s", self._data_key, temperature
+            )
```

Comment:
Its not ideal that the API would control the flow by returning `False` or raising an exception for both failure cases instead of raising an exception for either.

Reply:
You're right. It's not ideal. 
Returning `False` is legacy from the original idea of simply returning the success state. Retrying requests and reconnecting have all outgrown that concept at this point. In reality exception handling in the API needs an overhaul, but I expect that to take some time and not be ready until the next major release.  

This PR was an effort to improve the end user experience in the meantime.  If it's preferred that we leave things as-is until that overhaul is complete, we can kill this.

Reply:
I'm ok with this as an interim solution.

If you plan on refactoring the lib before the end of November it might be best to not merge this though so we don't have the churn of having to back it out later on when the lib is refactored.

Reply:
I don't think it will be complete by the end of November. The refactor interacts with a number of new features already in progress for the next major version of the lib and it's highly unlikely it will all be buttoned up by then.